File: server/src/Tables/JetpackInput.cs
namespace pillz.server.Tables;

[SpacetimeDB.Type]
public partial struct JetpackInput
{
    public float Fuel;
    public bool Enabled;
    public bool Throttling;
    
    public JetpackInput(uint fuel, bool enabled, bool throttling)
    {
        Fuel = fuel;
        Enabled = enabled;
        Throttling = throttling;
    }
}

File: server-rust/src/tables/jetpack_input.rs
use spacetimedb::SpacetimeType;
#[derive(SpacetimeType, Debug, Clone, Copy, PartialEq)]
pub struct JetpackInput {
    pub fuel: f32,
    pub enabled: bool,
    pub throttling: bool,
}

impl JetpackInput {
    pub fn new(fuel: f32, enabled: bool, throttling: bool) -> Self {
        Self {
            fuel,
            enabled,
            throttling,
        }
    }
}
